klog.go

```diff
@@ -1265,6 +1302,33 @@ func Infof(format string, args ...interface{}) {
 	logging.printf(infoLog, logging.logr, format, args...)
 }
 
+// InfoS structured logs to the INFO log.
+// The msg argument used to add constant description to the log line.
+// The key/value pairs would be join by "=" ; a newline is always appended.
+//
+// Basic examples:
+// >> klog.InfoS("Pod status updated", "pod", "kubedns", "status", "ready")
+// output:
+// >> I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kubedns" status="ready"
+func InfoS(msg string, keysAndValues ...interface{}) {
+	logging.printS(nil, logging.logr, msg, keysAndValues...)
+}
+
+func kvListFormat(b *bytes.Buffer, keysAndValues ...interface{}) {
+	for i := 0; i < len(keysAndValues); i += 2 {
+		var v interface{}
+		k, ok := keysAndValues[i].(string)
+		if !ok {
+			return
+		}
+		if i+1 < len(keysAndValues) {
```

Comment:
So if there is a dangling key, we just get nothing?  I think we'd be better off doing something like printf.  e.g. `key=nil` or `key=(MISSING)`

Reply:
act as key="(MISSING)"